components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2055,6 +2056,10 @@ private User getSCIMUser(String userName, List<String> claimURIList, Map<String,
             Map<String, String> attributes = SCIMCommonUtils.convertLocalToSCIMDialect(userClaimValues,
                     scimToLocalClaimsMap);
 
+            if (!attributes.containsKey(SCIMConstants.CommonSchemaConstants.ID_URI)) {
+                return scimUser;
+            }
```

Comment:
With this fix there's a possibility to return null from this method.
Make sure invocations of this method handles that

Reply:
it's already handled